addons/website_mass_mailing/static/src/js/website_mass_mailing.js

```diff
@@ -37,12 +34,6 @@ publicWidget.registry.subscribe = publicWidget.Widget.extend({
     start: function () {
         var self = this;
         var def = this._super.apply(this, arguments);
-        this.$popup = this.$target.closest('.o_newsletter_modal');
-        if (this.$popup.length) {
-            // No need to check whether the user subscribed or not if the input
-            // is in a popup as the popup won't open if he did subscribe.
```

Comment:
How is this handled now?

Reply:
This part is not handled anymore, I removed it figuring it wasn't that much of a need.
Worst case, after 7 days (cookie duration of the closed popup), you see the popup again.
Tell me if you prefer to re-introduce it.
I guess I can make the newsletter_form snippet add a data-is-subscribed on the snippet, and check inside an overide of the s_popup snippet when opening the popup, if the user is subscribed or not.

Reply:
I find that important. Like you said 7 days as chosen time seems a good idea and in that case I really not expect my users to have this popup appear every 7 days once they are subscribed.

> I guess I can make the newsletter_form snippet add a data-is-subscribed on the snippet, and check inside an overide of the s_popup snippet when opening the popup, if the user is subscribed or not.

And the main point is that it probably is not much to code indeed :+1: Maybe even more generically to handle any condition which may prevent a modal to be opened?

Reply:
I introduced a commit on top of the PR, that adapts the previous behavior to not show the modal if the user is already subscribed.